app/src/main/java/org/openobservatory/ooniprobe/fragment/PreferenceFragment.java

```diff
@@ -115,6 +115,8 @@ else if (preference instanceof EditTextPreference) {
                 getFragmentManager().beginTransaction().replace(android.R.id.content, newConcreteInstance(rootKey)).commit();
             }
         } else if (preference instanceof SwitchPreferenceCompat) {
+            if (key.equals(getString(R.string.max_runtime_enabled)))
+                return;
```

Comment:
I don't fully understand the logic here, can you explain @lorenzoPrimi?

Reply:
Because this code written by xan it's to manage the case of "not disable every category" and for some reasons it contrast with this new switch

Reply:
I still fail to fully understand, sorry 🤷‍♀

Reply:
I feel like we should not rely on hacks and understand how to better the code.

Reply:
It's not an hack, it's the only way to deal with this scenario "at least one enabled" with the `SwitchPreferenceCompat`
This is the algorithm explained: when disabling a category, check every other category, if at least one is enabled return true, else return false and disable this change.
If it's what you want I can move the next two lines in an external function.